porch/api/porch/v1alpha1/types_packagerevisions.go

```diff
@@ -78,6 +78,8 @@ type PackageRevisionSpec struct {
 
 	Lifecycle PackageRevisionLifecycle `json:"lifecycle,omitempty"`
 
+	DeletionProposed bool `json:"deletionProposed,omitempty"`
```

Comment:
I know we discussed reusing the `Lifecycle` field for this, but I had some concerns about that: https://github.com/GoogleContainerTools/kpt/issues/3264#issuecomment-1289978355

Reply:
Done.